client/js/widgets/scoreboard.js

```diff
@@ -0,0 +1,306 @@
+    return new Set([...super.readOnlyProperties(), '_totals']);
+  }
+
+  // Return a modified 'seat' array or object including the seat widgets (not just the seat ids) to actually be used.
+  getIncludedSeats() {
+    let seats = this.get('seats');
+    if(typeof seats == 'string') // Allow "seats": "Seat1"
+      seats = asArray(seats);
+    if(Array.isArray(seats) || seats === null) // Scoreboard just using seats
+       return [...widgetFilter(w => w.get('type') == 'seat' && (this.get('showAllSeats') || w.get('player')) && (!seats || seats.includes(w.get('id'))))]
+    else if(typeof seats == 'object') { // Scoreboard using teams
+      const newSeats = {};
+      for (const team in seats) {
+        newSeats[team] = asArray(seats[team]);
+        newSeats[team] = [... widgetFilter(w => w.get('type') == 'seat' && (this.get('showAllSeats') || w.get('player')) && newSeats[team].includes(w.get('id')))];
+      }
+      return newSeats;
+    } else // 'seats' property is not array or object, return null so table will be cleared.
+      return null;
+  }
+
+  getTotal(x) {
+    return asArray(x).reduce((partialSum, a) => partialSum + (parseFloat(a) || 0), 0)
+  }
+
+  addRowToTable(parent, values, cellType = 'td') {
+    const tr = parent.insertRow();
+    const v = asArray(values);
+    tr.innerHTML = Array(values.length).fill(`<${cellType}></${cellType}>`).join('');
+    for (let i=0; i < values.length; i++)
+      $a(cellType, tr)[i].innerText = values[i];
+    return tr;
+  }
+
+  // Compute desired text color from luminance of background
+  // See https://stackoverflow.com/questions/3942878
+  getFgColor(bgColor) {
+    function getHexColor(color) {
+      const a = document.createElement('div');
+      a.style.color = color;
+      const colorStr = window.getComputedStyle( document.body.appendChild(a) ).color;
+      const colors = colorStr.match(/\d+/g).map(function(a){ return parseInt(a,10); });
+      document.body.removeChild(a);
+      return colors;
+    }
+    const bg = getHexColor(bgColor);
+    return bg[0]*0.299 + bg[1]*0.587 + bg[2]*0.114 > 186 ? "#000000" : "#ffffff";
+  }
+
+  tableCreate(seats) {
+    /* This routine creates the HTML table for display in the scoreboard. It is
+     * complicated by the fact that the `seats` property can be either an array of
+     * seat IDs or an object whose keys are team names and each of whose values is an
+     * array of seat IDs.
+     * There are two major sections: the first computes the pScores array, which contains
+     * the array of scores, either for seats or for teams. The second section uses the
+     * pScores array to construct the HTML table.
+     * There are lots of other things going on, to get the totals line, round names, etc
+     * correct.
+     */
+
+    // First deal with cases where we just need to clear the table; invalid set of seats.
+    // We choose here to regard a result of [] or {} as a valid set of seats/teams with no entries.
+    if(seats===null) {
```

Comment:
Line 236 clears the table regardless, but only if there are no errors leading to an early return on line 225. It would be better (and less repeated code) if you unconditionally clear the table here so that if there are errors you are left with nothing rather than the previous table dom.